superagi/resource_manager/resource_manager.py

```diff
@@ -0,0 +1,147 @@
+import os
+
+from llama_index import SimpleDirectoryReader
+
+from superagi.config.config import get_config
+from superagi.helper.resource_helper import ResourceHelper
+from llama_index.schema import Document
+
+from superagi.lib.logger import logger
+from superagi.types.vector_store_types import VectorStoreType
+from llama_index.indices.response import ResponseMode
+
+
+class ResourceManager:
+    @classmethod
+    async def create_llama_document(cls, file_path: str = None, file_object=None):
+        """
+        Creates a document index from a given directory.
+        """
+
+        if file_path is None and file_object is None:
+            raise Exception("Either file_path or file_object must be provided")
+
+        if file_path is not None and file_object is not None:
+            raise Exception("Only one of file_path or file_object must be provided")
+
+        save_directory = ResourceHelper.get_root_input_dir() + "/"
+
+        if file_object is not None:
+            file_path = save_directory + file_object.filename
+            with open(file_path, "wb") as f:
+                contents = await file_object.read()
+                f.write(contents)
+                file_object.file.close()
+
+        documents = SimpleDirectoryReader(input_files=[file_path]).load_data()
+
+        if file_object is not None:
+            os.remove(file_path)
+
+        return documents
+
+    @classmethod
+    def generate_summary_of_document(cls, documents: list[Document], openai_api_key: str = None):
```

Comment:
We cant bind our methods to openai in case we change the model this method is totally use less.